platform-sdk/swirlds-virtualmap/src/main/java/com/swirlds/virtualmap/internal/cache/VirtualNodeCache.java

```diff
@@ -1508,78 +1516,82 @@ private static boolean dedupeByPath(
     /**
      * A mutation. Mutations are linked together within the mutation list. Each mutation
      * has a pointer to the next oldest mutation in the list.
-     *
-     * @param <U>
-     * 		The type of data held by the mutation.
+     * @param <K> The key type of data held by the mutation.
+     * @param <V> The type of data held by the mutation.
      */
-    private static final class Mutation<U> {
-        private volatile Mutation<U> next;
+    private static final class Mutation<K, V> {
```

Comment:
Can it be a record?

Reply:
I don't think it can without major rework. `record` classes are immutable, and `Mutation` has three mutable fields.